packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -848,42 +878,42 @@ export const createAzureCommunicationCallAdapter = async ({
 /**
  * @beta
  */
-export const createAzureCommunicationTeamsCallAdapter = async ({
+export const createTeamsCallAdapter = async ({
   userId,
   credential,
-  locator,
-  /* @conditional-compile-remove(PSTN-calls) */ alternateCallerId
-}: AzureCommunicationCallAdapterArgs): Promise<TeamsCallAdapter> => {
+  locator
+}: TeamsCallAdapterArgs): Promise<TeamsCallAdapter> => {
   const callClient = createStatefulCallClient({
-    userId,
-    /* @conditional-compile-remove(PSTN-calls) */ alternateCallerId
+    userId
   });
   const callAgent = await callClient.createTeamsCallAgent(credential, {
     undefined
   });
-  const adapter = createAzureCommunicationTeamsCallAdapterFromClient(callClient, callAgent, locator);
+  const adapter = createTeamsCallAdapterFromClient(callClient, callAgent, locator);
   return adapter;
 };
 
+type PartialArgsType<AdapterType> = AdapterType extends CallAdapter
+  ? Partial<AzureCommunicationCallAdapterArgs>
+  : Partial<TeamsCallAdapterArgs>;
+
+type AdapterTypeOf<CallType extends 'AzureCommunication' | 'Teams'> = CallType extends 'AzureCommunication'
+  ? CallAdapter
+  : TeamsCallAdapter;
+
 /**
  * @private
  */
 const useAzureCommunicationCallAdapterGeneric = <
-  AdapterType extends CallAdapter | /* @conditional-compile-remove(teams-identity-support) */ TeamsCallAdapter
+  CallType extends 'AzureCommunication' | 'Teams',
+  AdapterType extends AdapterTypeOf<CallType>
```

Comment:
we should avoid naming a type with a suffix of Type, where we use `AdapterType` can we just use `Adapter` instead?